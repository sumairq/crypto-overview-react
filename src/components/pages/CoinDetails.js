import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import SingleCoin from '../SingleCoin';
import PrintDetails from '../PrintDetails';
import { getDetailsFromApi } from '../../redux/coinDetails/coinDetails';

const CoinDetails = () => {
  const coin = useParams();
  const data = useSelector((state) => state.coinDetails[0]);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getDetailsFromApi(coin));
  }, []);

  return (
    <div>

      {(data)
        ? (
          <>
            <SingleCoin coin={data} vw="100vw" />
            <PrintDetails data={data} />
          </>
        ) : (
          <h1>loading</h1>
        )}

    </div>
  );
};

export default CoinDetails;
