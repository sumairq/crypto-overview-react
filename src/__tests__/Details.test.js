import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CoinDetails from '../components/pages/CoinDetails';
import store from '../redux/configureStore';

test('matches details component snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CoinDetails />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
