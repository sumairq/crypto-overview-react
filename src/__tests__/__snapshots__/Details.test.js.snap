// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches details component snapshot 1`] = `
<div>
  <h1>
    loading
  </h1>
</div>
`;
